go-controller/pkg/util/net_unit_test.go

```diff
@@ -278,3 +279,115 @@ func TestMatchIPFamily(t *testing.T) {
 		})
 	}
 }
+
+func TestJoinHostPortInt32(t *testing.T) {
+	tests := []struct {
+		desc    string
+		inpHost string
+		inpPort int32
+		outExp  string
+	}{
+		{
+			desc:    "empty string for host name and a negative port number",
+			inpPort: -25,
+			outExp:  ":-25",
+		},
+		{
+			desc:    "valid non-zero length string for host name",
+			inpHost: "hostname",
+			inpPort: 22,
+			outExp:  "hostname:22",
+		},
+		{
+			desc:    "valid IPv4 address as host name",
+			inpHost: "192.168.1.15",
+			inpPort: 22,
+			outExp:  "192.168.1.15:22",
+		},
+		{
+			desc:    "valid IPv6 address as host name",
+			inpHost: "fd01::1234",
+			inpPort: 22,
+			outExp:  "[fd01::1234]:22",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			res := JoinHostPortInt32(tc.inpHost, tc.inpPort)
+			t.Log(res)
+			assert.Equal(t, res, tc.outExp)
+		})
+	}
+}
+
+func TestIPAddrToHWAddr(t *testing.T) {
+	tests := []struct {
+		desc  string
+		inpIP net.IP
+	}{
+		// TODO: The function may need to be modified to ensure that the input net.IP passed is not empty.
+		// Passing empty net.IP causes panic
+		/*{
+			desc:	"test empty instance of net.IP",
+			inpIP: net.IP{},
+		},*/
+		{
```

Comment:
@vjayaramrh you can remove L328-333 since the assumption here is that the function parameter is going to be non-nil

Reply:
Well, this is an exported function and passing an empty net.IP{} is a valid input and should be handled gracefully, right? Should I add to the function documentation that passing an empty net.IP{} would panic?

Reply:
Well it is very common for an exported API to assume that the inputs are non-nil as long as the API caller knows about it. Yes, I agree that we should make it explicit. So, please go ahead with the function documentation.